### PR TITLE
Step 2 - architectural improvements

### PR DESCRIPTION
**Pull request by julichan**
Hello,
Here is a large server update:
- Convertion from CIJ to ES6 to make the app code more modern.
- Multi-database support architecture (I thought i could add mariadb and postgres later).
- Improved plugin and server setup handling.
- initialization and finalization logs.
- Dependencies overall updates!
- A fix on hidden errors.
- A fix on database life cycle to avoid potential corruption.
- Two fixes related to background upload possible malfunction (when the file already exists and when the insertion in database fails)

A small frontend update:
- dependencies updates

A small build update:
- dependencies updates
- node image update

This is a large enough PR so I did not change any logic.

### CODE REVIEW

File: server/plugins/db.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import fp from "fastify-plugin";
+import storesSetup from "../db/stores.js";
+
+export default fp(async function (fastify, opts, next) {
+  // Define which database we want and initilize it
+  // For now only sqlite is available
+
+  try {
+    const databasePluginName = opts.databaseType || "sqlite";
+    const databasePluginFile = `../db/${databasePluginName}/database.js`;
```
Comment (AlexSciFier) [inline server/plugins/db.js:10]: This file doesn't exist

File: server/package.json
Comment (AlexSciFier) [inline server/package.json:1]: It gives an error when i tried to run `npm run dev`
```
TypeError: Cannot read properties of undefined (reading 'length')
    at isInvalidAsyncPlugin (C:\Users\Alexandr\AppData\Roaming\npm\node_modules\fastify-cli\util.js:45:17)
    at requireServerPluginFromPath (C:\Users\Alexandr\AppData\Roaming\npm\node_modules\fastify-cli\util.js:82:7)
    at async runFastify (C:\Users\Alexandr\AppData\Roaming\npm\node_modules\fastify-cli\start.js:89:12)
```
Maybe scripts need to be changed?

File: server/plugins/db.js (hunk 1 above)
Comment (AlexSciFier) [reply server/plugins/db.js:10]:
```
Database plugin initialization completed.
Error: Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\Dev\neonlink-fork\neonlink\server\db\sqlite\database.js' imported from D:\Dev\neonlink-fork\neonlink\server\plugins\db.js
```

**Review by AlexSciFier — CHANGES_REQUESTED** [review]

File: server/package.json
Comment (julichan) [reply server/package.json:1]: No, it s missing some files. Please wait. The problem is that i created a folder that was inside gitignore and did not notice. However, i continued making changes behind and those files are now too evolved for this Merge request version. I fixed the problem but i need to finish my current development for it to work.
I'll make another merge request soon enough.

Comment (AlexSciFier) [reply server/package.json:1]: Oh, ok! I'll wait. Anyway you doing a good job. I get more free times, so i think i can start on adding multiusers function.

Comment (julichan) [reply server/package.json:1]: Wait before starting on the multi user function. i was planning to move on to this right after this but the functionalities as they are now cannot be ported multi users. I believe some features must be updated first. that's why i'm rewriting the server without changing its feature for now.
I think we should have a chat over what features the multi user function will impact.
For exemple, we have the noLogin feature. This should be either removed or implemented differently. In my vision, i think keeping it is possible if we handle the uuid thing differently. the nologin mode should keep its settings purely in the browser local storage so many users can use the app whilst sharing links and dashboards without having an actual user and allowing each users to have their own preference.
I think we should have a default user with id 0 which either serve as default user for nologin mode or as a shared user reference for elements shared accross user such as shared links and dashboards whilst retaining private ones using their own user as reference. In multi user mode, the prevailing settings should be the one stored on server and that would make the sync option useless whilst making the app upgrade so much easier.
But all this is pure thoughts and I believe you and me see the multi user feature differently which is why we should discuss it. :) 
I've so far completely revamped the server and fixed few bugs along the way. The server is acting much faster thanks to the libs updates and the changes makes it much easier to update. A merge request should be coming tonight as it's going. 
However, I haven't done much on the client side. I'm not too knowledgable about reactjs but my vue js experience help me a little so i shouldn't be too lost. 

Amongst features that i think are required for the multi user part,  i m thinking of altering the uuid logic, adding a request context and a task scheduler. 

What are your ideas ?

**Pull request closed without merging**